clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -3842,14 +3845,67 @@ private GenericRecord readStoredValueRecord(
    * Clone DIV check results from OffsetRecord to the DIV validator that is used for leader consumption thread.
    *
    * This step is necessary in case leadership handovers happen during batch push; if this node is used to be a
-   * follower replica for a partition and it's promoted to leader in the middle of a batch push, leader DIV validator
+   * follower replica for a partition, and it's promoted to leader in the middle of a batch push, leader DIV validator
    * should inherit/clone the latest producer states from the DIV validator in drainer; otherwise, leader will encounter
    * MISSING message DIV error immediately.
    */
-  private void restoreProducerStatesForLeaderConsumption(int partition) {
-    consumerDiv.clearPartition(partition);
-    cloneProducerStates(partition, consumerDiv);
-    // TODO: the Global RT DIV version needs to be implemented in a future PR
+  void restoreProducerStatesForLeaderConsumption(int partition) {
+    getConsumerDiv().clearPartition(partition);
+    if (isGlobalRtDivEnabled()) {
+      loadGlobalRtDiv(partition);
+    } else {
+      cloneDrainerDivProducerStates(partition, getConsumerDiv());
+    }
+  }
+
+  void loadGlobalRtDiv(int partition) {
+    getKafkaClusterIdToUrlMap().forEach((clusterId, brokerUrl) -> {
+      loadGlobalRtDiv(partition, brokerUrl);
+    });
+  }
+
+  /**
+   * Load the stored Global RT DIV object from the storage engine into the Consumer DIV during state transition
+   * to LEADER. The RT DIV needs to be loaded per-broker url, and the latest consumed RT offset needs to be updated.
+   */
+  void loadGlobalRtDiv(int partition, String brokerUrl) {
+    PartitionConsumptionState pcs = getPartitionConsumptionState(partition);
+    final PubSubTopic topic = pcs.getOffsetRecord().getLeaderTopic(getPubSubTopicRepository());
+    final PubSubTopicPartition topicPartition = new PubSubTopicPartitionImpl(topic, pcs.getPartition());
+
+    String globalRtDivKey = getGlobalRtDivKeyName(brokerUrl);
+    byte[] keyBytes = globalRtDivKey.getBytes();
+    final ChunkedValueManifestContainer valueManifestContainer = new ChunkedValueManifestContainer();
+    GenericRecord valueRecord = readStoredValueRecord(
+        pcs,
+        keyBytes,
+        AvroProtocolDefinition.GLOBAL_RT_DIV_STATE.getCurrentProtocolVersion(),
+        topicPartition,
+        valueManifestContainer);
+    if (valueRecord == null) {
+      if (pcs.getLeaderOffset(brokerUrl, pubSubTopicRepository, false) > 0) {
+        LOGGER.warn(
```

Comment:
Could we also printout the value of `getLeaderOffset` for the ease of debugging?